File: Julia/Problem012.jl
function getNumDivisors(num)
	if num == 0
		return 0
	end

	divisors = zeros(0)
	i = 1

	while i * i <= num
		if num % i == 0
			if num / i == i
				append!(divisors, i)
			else
				append!(divisors, i)
				append!(divisors, Int(num / i))
			end
		end
		i += 1
	end

	return length(divisors)
end


# Parameters:
minDivs = 500
###############

count = 1
triNum = 0

while getNumDivisors(triNum) <= minDivs
	global triNum += count
	global count += 1
end

println(triNum)

File: C++/Problem012.cpp
#include <iostream>
#include <vector>


unsigned long int numDivisors(unsigned long int n) {
	if (n == 0) {
		return 0;
	}

	unsigned long int divisors {}, i {1};

	while (i * i <= n) {
		if (n % i == 0) {
			if (n / i == i)	divisors++;
			else			divisors += 2;
		}

		i++;
	}

	return divisors;
}


int main() {
	// Parameters:
	int minDivs = 500;
	///////////////

	unsigned long int count {1}, triNum {0};

	while (numDivisors(triNum) < minDivs) {
		triNum += count;
		count++;
	}

	std::cout << triNum << std::endl;

	return 0;
}